ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -215,7 +258,7 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
                                         </div>
                                         <div className='application-details__tree'>
                                             {refreshing && <p className='application-details__refreshing-label'>Refreshing</p>}
-                                            {((pref.view === 'tree' || pref.view === 'network') && (
+                                            {((pref.view === 'tree' || pref.view === 'network' || pref.view === 'compact') && (
```

Comment:
This is speaking to the design of this solution but I wonder if instead of having a separate view, you could use the same current Tree view and just add a toggle to collapse and expand the groups. Might want to check with Alex regarding this to see what he prefers.

Reply:
I agree, when there's nothing to group, the Compact view will look exactly as the Tree view. For now it's easier and cleaner to have the Compact view as same level as other views, we can discuss further with the UI team.

Reply:
I agree with @keithchong . I think from the user's perspective "compact" view if is a toggle that expands/collapse groups. I don't think it is a blocker for this PR and can be done an as improvement after further discussion.